libs/@local/hql/README.md

```diff
@@ -0,0 +1,51 @@
+# HQL - HASH Query Language
+
+This is the compiler, planner and executor for the HASH Query Language (HQL). HQL is a functional programming language that is created using a subset of JSON.
+
+## Language
+
+Language design documentation and a specification of the language will be provided in the future.
+
+The CST of the language is very basic, only consisting of three different constructs:
+
+- Function Invocation: A function invocation has multiple possible forms:
+  - `[name, ...args]`
+  - `{"fn": name, "args": args}`
+- Constants: Constants are simple values that are not functions, they are defined using:
+  - `{"const": value}`, this will convert the JSON type to the corresponding `hql` type.
+  - `{"const": value, "type": type}`, this will convert the JSON type to the specified `hql` type (if possible), if not possible this will result in an error.
+- Variables: Variables are references to values that are defined elsewhere, they are defined using:
+  - `"symbol"` (any string is a reference to a variable)
+  - `{"var": name}` (this is a more explicit way of defining a variable)
+- Signature: Signature is a special type, which is parsed independently of variables, they have a specific form (specified in the CST crate) and are defined using:
+  - `{"sig": ...}`
+  - `"sig"`
```

Comment:
What does `"sig"` represent?

Reply:
A signature string that follows the signature EBNF.

Reply:
I cannot follow, why the signature would not be parsed into the CST?

Reply:
no, it will be, it's:

```
"() -> Unit"
```

that is then parsed into the CST as a signature.

Reply:
which is different to e.g. `"abc"`, which would instead refer to a path, which either points to a type or a variable.

Reply:
I think this is highly confusing. Can we please only use a single representation for the exact same thing?